Lib/tarfile.py

```diff
@@ -1669,21 +1670,21 @@ def gzopen(cls, name, mode="r", fileobj=None, compresslevel=9, **kwargs):
         try:
             from gzip import GzipFile
         except ImportError:
-            raise CompressionError("gzip module is not available")
+            raise CompressionError("gzip module is not available") from None
 
         try:
             fileobj = GzipFile(name, mode + "b", compresslevel, fileobj)
-        except OSError:
+        except OSError as e:
```

Comment:
Should not it be more specific BadGzipFile and additionally EOFError?

Reply:
I don't think so -- both of those are more specific and more informative than `ReadError`.